### PR TITLE
fix[layout]: row id layout reader cannot stop if inner layout registering its splits

### PR DESCRIPTION
**Pull request by joseph-isaacs**

### CODE REVIEW

File: vortex-layout/src/layouts/row_idx/mod.rs [hunk 1]
```diff
@@ -131,12 +131,7 @@ impl LayoutReader for RowIdxLayoutReader {
         row_offset: u64,
         splits: &mut BTreeSet<u64>,
     ) -> VortexResult<()> {
-        // Since RowIdx isn't a field, we only need to register splits for the child layout
-        // if there are any fields in the mask at all.
-        if !field_mask.is_empty() {
-            self.child.register_splits(field_mask, row_offset, splits)?;
-        }
-        Ok(())
```
Comment (a10y) [inline vortex-layout/src/layouts/row_idx/mod.rs:139]: can you say more why this is wrong?

**Review by robert3005 — APPROVED** [review]

**Comment by robert3005** [comment]
But probably need to have a test for empty projection and no filter with row id

File: vortex-layout/src/layouts/row_idx/mod.rs (hunk 1 above)
Comment (joseph-isaacs) [reply vortex-layout/src/layouts/row_idx/mod.rs:139]: Cause we never ask the child layout for its splits

Comment (robert3005) [reply vortex-layout/src/layouts/row_idx/mod.rs:139]: Empty mask doesn't mean we reading no rows, just that the rows have 0 fields/cells
